modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/InfrastructureElementModelDelegate.java

```diff
@@ -155,8 +155,50 @@ public void visitObjectValue(ParameterObjectValue objectValue) {
 
   }
 
+  private void createReconnectionConfig(ParameterValue value,
+                                        ParameterModel parameterModel,
+                                        DslElementSyntax paramDsl,
+                                        ComponentConfiguration.Builder parentConfig,
+                                        DslElementModel.Builder parentElement) {
+
+    ComponentConfiguration.Builder config = ComponentConfiguration.builder()
+        .withIdentifier(builder()
+            .namespace(CORE_PREFIX)
+            .name(RECONNECTION_CONFIG_PARAMETER_NAME)
+            .build());
+
+    final DslElementModel.Builder<Object> elementBuilder = DslElementModel.builder()
+        .withModel(parameterModel)
+        .withDsl(paramDsl);
+
+    ((ParameterObjectValue) value).getParameters()
+        .forEach((name, fieldValue) -> fieldValue.accept(new ParameterValueVisitor() {
+
+          @Override
+          public void visitSimpleValue(ParameterSimpleValue text) {
+            config.withParameter(name, text.getValue());
+          }
+
+          @Override
+          public void visitObjectValue(ParameterObjectValue objectValue) {
+            if (name.equals(RECONNECTION_STRATEGY_PARAMETER_NAME)) {
+              createReconnectionStrategy(fieldValue,
+                                         ((ObjectType) parameterModel.getType())
+                                             .getFieldByName(RECONNECTION_STRATEGY_PARAMETER_NAME).get(),
+                                         paramDsl.getContainedElement(RECONNECTION_STRATEGY_PARAMETER_NAME).get(),
+                                         config, elementBuilder);
+            }
+          }
+        }));
+
+    final ComponentConfiguration result = config.build();
+    parentConfig.withNestedComponent(result);
+    parentElement.containing(elementBuilder.withConfig(result).build());
+
+  }
+
   private void createReconnectionStrategy(ParameterValue value,
-                                          ParameterModel parameterModel,
+                                          Object parameterModel,
```

Comment:
technical limitations I already discussed with Marra and will have to tackle separately. Basically this can either be a parameterModel or a declaration